options/config/common.go

```diff
@@ -9,19 +9,22 @@ import (
 	"github.com/plgd-dev/go-coap/v3/message/pool"
 	"github.com/plgd-dev/go-coap/v3/net/blockwise"
 	"github.com/plgd-dev/go-coap/v3/net/client"
+	"github.com/plgd-dev/go-coap/v3/net/responsewriter"
 	"github.com/plgd-dev/go-coap/v3/pkg/runner/periodic"
 )
 
 type ErrorFunc = func(error)
 
-type GoPoolFunc = func(func()) error
+type GoPoolFunc[C responsewriter.Client] func(
+	processReqFunc func(req *pool.Message, cc C, handler func(w *responsewriter.ResponseWriter[C], r *pool.Message)),
```

Comment:
Three embedded `func` declarations are a bit tough to read, maybe add `type = func(...)` aliases?

Reply:
Fixed